helix-view/Cargo.toml

```diff
@@ -33,6 +33,8 @@ slotmap = "1"
 
 encoding_rs = "0.8"
 chardetng = "0.1"
+similar = "1.3"
+ropey = "1.3"
```

Comment:
Can you move most of the code into core? The diffing and generating a transaction shouldn't be tied to Document, it'll be easier to test that way. The method on the document can then simply wrap the function from core and apply the resulting transaction